stdlib/public/Concurrency/Executor.swift

```diff
@@ -370,17 +698,33 @@ public struct UnownedSerialExecutor: Sendable {
     unsafe self.executor = Builtin.buildComplexEqualitySerialExecutorRef(executor)
   }
 
+  /// Automatically opt-in to complex equality semantics if the Executor
+  /// implements `Equatable`.
+  @available(SwiftStdlib 6.2, *)
+  @inlinable
+  public init<E: SerialExecutor>(_ executor: __shared E) {
+    if executor._isComplexEquality {
+      unsafe self.executor = Builtin.buildComplexEqualitySerialExecutorRef(executor)
+    } else {
+      unsafe self.executor = Builtin.buildOrdinarySerialExecutorRef(executor)
+    }
+  }
```

Comment:
Aha, got it -- that's the bit I was wondering about up-thread; This is quite nice, I think it's ok to use this new style of opting in, very nice! 👍

Reply:
Does this mean though that we should deprecate the other old initializers?